ipa-core/src/hpke/mod.rs

```diff
@@ -100,10 +100,9 @@ pub fn open_in_place<'a, R: PrivateKeyRegistry>(
     key_registry: &R,
     enc: &[u8],
     ciphertext: &'a mut [u8],
-    info: &Info,
+    key_id: u8,
```

Comment:
Previously we used key id to authenticate ct and to pick the appropriate private key from `key_registry`. Based on this change, I think we should no longer take `key_registry` and instead just take `&PrivateKey` parameter without `key_id`. I don't remember why we decided to authenticate the cipher with key id, but I agree this doesn't seem necessary

Reply:
this function is shared with the OprfReport implementation. Is it okay to alter that?

Reply:
I think it should be ok to change this, given that we need to update `OprfReport` to call the new version of it anyway